src/lib.rs

```diff
@@ -266,29 +290,47 @@ impl Config {
         self
     }
 
-    fn is_flag_supported(&mut self, flag: &str) -> io::Result<bool> {
+    fn is_flag_supported(&mut self, flag: &str) -> Result<bool, Error> {
```

Comment:
I think the original source of the error here is `io::Error`, so perhaps this could continue to return a `io::Result`?